src/Microsoft.Security.Utilities.Core/HighPerformanceScanner.cs

```diff
@@ -28,25 +29,48 @@ namespace Microsoft.Security.Utilities;
 /// </summary>
 internal sealed class HighPerformanceScanner
 {
+    private Dictionary<int, CompiledHighPerformancePattern> _patternsByPackedSignature = new();
+
 #if NET9_0_OR_GREATER
-    private SearchValues<string> _signatures;
+    private static SearchValues<string> s_emptySignatures = SearchValues.Create(Array.Empty<string>(), StringComparison.Ordinal);
+    private SearchValues<string> _signatures = s_emptySignatures;
 #elif NET8_0_OR_GREATER
-    private SearchValues<char> _signatureStarts;
+    private static SearchValues<char> s_emptySignatureStarts = SearchValues.Create(Array.Empty<char>());
+    private SearchValues<char> _signatureStarts = s_emptySignatureStarts;
 #else
-    private char[] _signatureStarts;
+    private char[] _signatureStarts = Array.Empty<char>();
 #endif
-    private Dictionary<int, CompiledHighPerformancePattern> _patternsByPackedSignature;
+
+    public HighPerformanceScanner()
+    {
+    }
 
     public HighPerformanceScanner(IEnumerable<CompiledHighPerformancePattern> patterns)
```

Comment:
>HighPerformanceScanner [](http://example.com/codeflow?start=11&length=22)

Here's something small to discuss, we could simply make this ctor have an optionally available parameter. on doing this, though, the class loses its parameterless constructor, which has impact for generics, at least.

Any thoughts on this? SEV5 just discussing in general.

Reply:
Yeah, that would be fine here since this is an internal type. I'll leave it only because this PR is getting big.